File: src/main/java/com/hackclub/hccore/commands/DiscordCommand.java
package com.hackclub.hccore.commands;

import com.hackclub.hccore.HCCorePlugin;

import com.comphenix.protocol.PacketType;
import com.hackclub.hccore.HCCorePlugin;
import com.hackclub.hccore.discord.DiscordBot;
import net.md_5.bungee.api.chat.ClickEvent;
import net.md_5.bungee.api.chat.TextComponent;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import java.util.List;


public class DiscordCommand implements CommandExecutor {
    private final HCCorePlugin plugin;

    public DiscordCommand(HCCorePlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        String inviteLink = plugin.getConfig().getString("discord-invite-url");

        TextComponent component = new TextComponent("Join our Discord Server: ");
        component.setColor(ChatColor.GREEN.asBungee());

        TextComponent inviteComponent = new TextComponent(inviteLink);
        inviteComponent.setColor(ChatColor.BLUE.asBungee());
        inviteComponent.setClickEvent(new ClickEvent(ClickEvent.Action.OPEN_URL, inviteLink));
        inviteComponent.setUnderlined(true);
        sender.spigot().sendMessage(component, inviteComponent);

        return true;
    }
}

File: src/main/java/com/hackclub/hccore/commands/DiscordCommand.kt
package com.hackclub.hccore.commandsimport

import net.md_5.bungee.api.chat.ClickEvent
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

com.hackclub.hccore.HCCorePlugin
import com.hackclub.hccore.PlayerData
import com.hackclub.hccore.tasks.CheckAdAstraTask
import com.hackclub.hccore.utils.gson.GsonUtil
import com.hackclub.hccore.utils.gson.LocationSerializer
import com.hackclub.hccore.utils.gson.LocationMapDeserializer
import com.hackclub.hccore.utils.gson.LocationMapSerializer
import com.hackclub.hccore.utils.gson.LocationDeserializer
import java.util.LinkedHashMap
import com.hackclub.hccore.events.player.PlayerAFKStatusChangeEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.entities.TextChannel
import net.dv8tion.jda.api.events.ReadyEvent
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
import net.dv8tion.jda.api.JDABuilder
import javax.security.auth.login.LoginException
import java.lang.Exception
import com.hackclub.hccore.commands.ColorCommand
import java.util.stream.Collectors
import com.hackclub.hccore.commands.StatsCommand
import java.text.SimpleDateFormat
import com.hackclub.hccore.commands.UpvoteCommand
import com.hackclub.hccore.commands.DownvoteCommand
import com.hackclub.hccore.commands.TableflipCommand
import java.lang.Runnable
import com.hackclub.hccore.listeners.SleepListener
import com.comphenix.protocol.events.ListenerPriority
import com.comphenix.protocol.PacketType
import com.comphenix.protocol.events.PacketAdapter
import com.comphenix.protocol.events.PacketEvent
import com.comphenix.protocol.wrappers.EnumWrappers
import com.comphenix.protocol.wrappers.WrappedGameProfile
import org.bukkit.craftbukkit.v1_17_R1.advancement.CraftAdvancement
import net.minecraft.advancements.AdvancementDisplay
import net.minecraft.network.chat.IChatBaseComponent
import net.minecraft.advancements.AdvancementFrameType
import de.tr7zw.changeme.nbtapi.NBTTileEntity
import java.io.FileReader
import java.io.IOException
import java.io.FileWriter
import java.util.UUID
import java.util.HashMap
import com.hackclub.hccore.DataManager
import com.comphenix.protocol.ProtocolManager
import com.hackclub.hccore.discord.DiscordBot
import com.comphenix.protocol.ProtocolLibrary
import com.hackclub.hccore.commands.AFKCommand
import com.hackclub.hccore.commands.LocCommand
import com.hackclub.hccore.commands.NickCommand
import com.hackclub.hccore.commands.PingCommand
import com.hackclub.hccore.commands.ShrugCommand
import com.hackclub.hccore.commands.SpawnCommand
import com.hackclub.hccore.commands.DiscordCommand
import com.hackclub.hccore.listeners.AdvancementListener
import com.hackclub.hccore.listeners.AFKListener
import com.hackclub.hccore.listeners.BeehiveInteractionListener
import com.hackclub.hccore.listeners.PlayerListener
import com.hackclub.hccore.listeners.NameChangeListener
import com.hackclub.hccore.tasks.AutoAFKTask
import hu.trigary.advancementcreator.AdvancementFactory
import hu.trigary.advancementcreator.shared.ItemObject

class DiscordCommand(plugin: HCCorePlugin) : CommandExecutor {
    private val plugin: HCCorePlugin
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        val inviteLink: String = plugin.getConfig().getString("discord-invite-url")
        val component = TextComponent("Join our Discord Server: ")
        component.color = ChatColor.GREEN.asBungee()
        val inviteComponent = TextComponent(inviteLink)
        inviteComponent.color = ChatColor.BLUE.asBungee()
        inviteComponent.clickEvent = ClickEvent(ClickEvent.Action.OPEN_URL, inviteLink)
        inviteComponent.setUnderlined(true)
        sender.spigot().sendMessage(component, inviteComponent)
        return true
    }

    init {
        this.plugin = plugin
    }
}